src/v2/Apps/Algolia/routes/AlgoliaResults/AlgoliaResults.tsx

```diff
@@ -0,0 +1,124 @@
+import { useState, useMemo, useRef, useEffect } from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import { AlgoliaResults_system } from "v2/__generated__/AlgoliaResults_system.graphql"
+import algoliasearch from "algoliasearch"
+import {
+  InstantSearch,
+  Configure,
+  connectPagination,
+  connectStateResults,
+} from "react-instantsearch-dom"
+import { useRouter } from "v2/System/Router/useRouter"
+import { AlgoliaIndicesFragmentContainer } from "./Components/AlgoliaIndices"
+import { AlgoliaPagination } from "./Components/AlgoliaPagination"
+import {
+  ANCHOR_CONTAINER_ID,
+  HITS_PER_PAGE,
+  DEBOUNCE_TIME,
+} from "../../constants"
+import { Box } from "@artsy/palette"
+import { createURL, searchStateToUrl, urlToSearchState } from "../../Utils/url"
+import { AlgoliaResultsListFragmentContainer } from "./Components/AlgoliaResultsList"
+import { AlgoliaResultsMeta } from "./Components/AlgoliaResultsMeta"
+
+interface AlgoliaResultsProps {
+  system: AlgoliaResults_system
+}
+
+const ConnectedAlgoliaPagination = connectPagination(AlgoliaPagination)
+const ConnectedAlgoliaResults = connectStateResults(
+  AlgoliaResultsListFragmentContainer
+)
+
+export const AlgoliaResults: React.FC<AlgoliaResultsProps> = ({ system }) => {
+  const { algolia } = system
+  const { match } = useRouter()
+  const [searchState, setSearchState] = useState(
+    urlToSearchState(match.location.search)
+  )
+  const [selectedTabIndex, setSelectedTabIndex] = useState(0)
+  const debouncedSetStateRef = useRef<NodeJS.Timeout | null>(null)
+  const searchClient = useMemo(() => {
+    if (algolia?.appID && algolia.apiKey) {
+      return algoliasearch(algolia.appID, algolia.apiKey)
+    }
+
+    return null
+  }, [algolia?.appID, algolia?.apiKey])
+
+  const onSearchStateChange = updatedSearchState => {
+    if (debouncedSetStateRef.current) {
+      clearTimeout(debouncedSetStateRef.current)
+    }
+
+    debouncedSetStateRef.current = setTimeout(() => {
+      const params = searchStateToUrl(updatedSearchState)
+      const url = `${match.location.pathname}${params}`
+
+      window.history.pushState(null, "", url)
```

Comment:
We'll want to use our router here -- `const router = useRouter()` (from v2/System/Router`)

Reply:
I initially tried using `router.push`, but unfortunately I got the wrong behavior: when using pagination, a new page always opens from the top and there is no smooth scrolling animation (see attached video)

https://user-images.githubusercontent.com/3513494/147920161-9a023d6d-7061-4798-8b64-a53068b5326e.mp4